source/renderer/app/components/staking/stake-pools/StakePoolsTable.hooks.tsx

```diff
@@ -0,0 +1,324 @@
+import React, { useMemo } from 'react';
+import { orderBy } from 'lodash';
+import classNames from 'classnames';
+import BigNumber from 'bignumber.js';
+import moment from 'moment';
+import { FormattedHTMLMessage } from 'react-intl';
+import { PopOver } from 'react-polymorph/lib/components/PopOver';
+import { PoolPopOver } from '../widgets/PoolPopOver';
+import { Intl } from '../../../types/i18nTypes';
+
+import styles from './StakePoolsTable.scss';
+import StakePool from '../../../domains/StakePool';
+import { getColorFromRange, getSaturationColor } from '../../../utils/colors';
+import {
+  formattedWalletAmount,
+  toFixedUserFormat,
+} from '../../../utils/formatters';
+import { messages } from './StakePoolsTable.messages';
+
+export const defaultTableOrdering = {
+  ranking: 'asc',
+  ticker: 'asc',
+  saturation: 'asc',
+  cost: 'asc',
+  profitMargin: 'asc',
+  producedBlocks: 'desc',
+  nonMyopicMemberRewards: 'desc',
+  pledge: 'asc',
+  retiring: 'asc',
+};
+
+interface UseSortedStakePoolListArgs {
+  stakePoolList: StakePool[];
+  sortBy: string;
+  order: 'asc' | 'desc';
+}
+
+export const useSortedStakePoolList = ({
+  stakePoolList,
+  sortBy,
+  order,
+}: UseSortedStakePoolListArgs) =>
+  useMemo(
+    () =>
+      orderBy(
+        stakePoolList.map((stakePool) => {
+          let calculatedPledge;
+          let calculatedCost;
+          let formattedTicker;
+
+          if (sortBy === 'ticker') {
+            formattedTicker = stakePool.ticker
+              .replace(/[^\w\s]/gi, '')
+              .toLowerCase();
+          }
+
+          if (sortBy === 'pledge') {
+            const formattedPledgeValue = stakePool.pledge.toFixed(2);
+            calculatedPledge = Number(
+              parseFloat(formattedPledgeValue).toFixed(2)
+            );
+          }
+
+          if (sortBy === 'cost') {
+            const formattedCostValue = stakePool.cost.toFixed(2);
+            calculatedCost = Number(parseFloat(formattedCostValue).toFixed(2));
+          }
+
+          return {
+            ...stakePool,
+            calculatedPledge,
+            calculatedCost,
+            formattedTicker,
+          };
+        }),
+        ['formattedTicker', 'calculatedPledge', 'calculatedCost', sortBy],
+        [order, order, order, order]
+      ),
+    [stakePoolList, order, sortBy]
+  );
+
+type UseCreateColumnsArgs = {
+  currentTheme: string;
+  showWithSelectButton?: boolean;
+  onSelect?: (...args: Array<any>) => any;
+  containerClassName: string;
+  numberOfRankedStakePools: number;
+  selectedPoolId?: string;
+  onOpenExternalLink: (...args: Array<any>) => any;
+  intl: Intl;
+};
+
+export const useCreateColumns = ({
+  numberOfRankedStakePools,
+  intl,
+  currentTheme,
+  onOpenExternalLink,
+  onSelect,
+  selectedPoolId,
+  containerClassName,
+  showWithSelectButton,
+}: UseCreateColumnsArgs) =>
+  useMemo(
```

Comment:
Should the useMemo have some args in the dependency array?

Reply:
I added the property `numberOfRankedStakePools` to fix the issue reported by @gabriela-ponce https://github.com/input-output-hk/daedalus/pull/2924#issuecomment-1070911680